### ISSUE TITLE
Suggestion for streamlining TSC meetings

### ISSUE THREAD

**Issue by ilyavolodin** [eslint/tsc-meetings#5]
I think we can improve the process of discussing issues that are marked as `tsc-agenda`. We spent too much time going into the details of implementation on at least two of those issues. And while sometime it's necessary, I think we can do better. Here are my suggestions for future meetings:

* For every discussed issue it would be ideal to have short summary of what is being discussed. I know that requires more time from meeting organizer upfront, but I think it will be less time overall then length discussion during the meeting.
* After short summary is provided, tsc members should vote to accept, decline or no opinion on the proposed change (without discussion).
* If everyone wants to accept or decline the change, we can move on to the next issue.
* If most people have no opinion or the vote is split, we can discuss the issue in more details.
* After the discussion is complete, or after 15 minutes tsc members can vote again, if the vote is still split, the issue is postponed till the next meeting.
* During discussion, try to avoid implementation details, unless they are critical to acceptance of the issue.

**Comment by nzakas** [comment]
I agree that we spent too much time on implementation details. We should definitely stick to deciding on ideas wherever possible and table implementation discussion whenever possible. Some other thoughts:

> For every discussed issue it would be ideal to have short summary of what is being discussed. I know that requires more time from meeting organizer upfront, but I think it will be less time overall then length discussion during the meeting.

What if the person who adds the "TSC agenda" label must also write a comment with a summary that explains what they want to resolve? That way, the info is right there in the issue and doesn't (unfairly) become the responsibility of the moderator. 

> After short summary is provided, tsc members should vote to accept, decline or no opinion on the proposed change (without discussion).

Voting is only done as a last resort (per our [governance model](http://eslint.org/docs/maintainer-guide/governance.html#consensus-seeking-process). The better way to accomplish this is to say something like, "the proposal is x, does anyone object?" If there are no objections, then we are done. If there are, then we discuss and see what the objection is 

I'm a bit uncomfortable with "without discussion " as a recommendation. We need to make sure everyone understands what is being discussed before we ask their opinion. We should be allowed to ask questions and discuss where necessary. 

I'm also not comfortable limiting discussions to 15 minutes. We should just push through and finish or decide that we need more info, but we shouldn't cut off just because it's taken 15 minutes. 

Overall, I think the biggest things we can do are eliminate unnecessary implementation details from the conversation and make sure we have good summaries for each issue. (Also, if everyone looked through the agenda items the day before rather than waiting until the meeting.)

**Comment by ilyavolodin** [comment]
By voting I mean wanting to see if people don't feel the same about the issue. I feel like the first issue we discussed, we spent 25 minutes talking about something that everyone agreed on from the begging. It would be nice to see where everyone stands before we start the discussion.
And I also don't mean no discussion allowed initially, if somebody is not clear on what is being proposed, they should ask questions. Summary would also help with making it clear what is being discussed. I, however, don't think we should start with "why do/don't you think it's a good idea?" questions.
Having person who marks the issue as `tsc agenda` create a short summary is a good idea. That works for me.

**Comment by alberto** [comment]
> What if the person who adds the "TSC agenda" label must also write a comment with a summary that explains what they want to resolve? That way, the info is right there in the issue and doesn't (unfairly) become the responsibility of the moderator.

Good idea, I support that.

> We need to make sure everyone understands what is being discussed before we ask their opinion. We should be allowed to ask questions and discuss where necessary.

Agree, we should present the issue and see if anybody needs any clarification before voting.

> I'm also not comfortable limiting discussions to 15 minutes. We should just push through and finish or decide that we need more info, but we shouldn't cut off just because it's taken 15 minutes.

I agree we don't want to rush or limit discussions. For the same reason, I also think it would be valuable to have some mechanism to defer an issue, to leave time for reflection.

**Comment by alberto** [comment]
My 2 cents.

- We tackled a good amount of issues, so yay for us! 👏  

- I think the first issue took longer because, well, it was the first one. :) The next ones went smoother, so let's give us some time to self-adapt before worrying too much, anyways.

- We should put less issues in the agenda next time.

- I think we should try to be stricter with timeboxing. I'm ok if we decide to have longer meetings at first (maybe 90 min?) until we clear our stack a little. There will probably less issues in the future. But I think it's better for everyone (tsc members and watchers) to be upfront about the duration. I'm sure no one wants to leave in the middle of the meeting. After all, we'll meet again in two weeks. If there is something so important that we feel it can't wait until the next meeting, we should say so beforehand and ensure we discuss that first.

- I'm going to open issues for discussion of the topics I suggested, so they are not so open ended.

- @nzakas did taking notes slow you down? Maybe we can relieve you from that and someone else can do it. Or we could use a shared doc.

**Comment by nzakas** [comment]
@ilyavolodin I get the idea behind voting, but that's a very inefficient way of doing things. The consensus -seeking process moves a lot faster. I think if we can get a good summary and then call for objections that we can achieve what you want without the overhead of voting. 

@alberto agreed on timing, this week was hard because we had never done it before. I think limiting to 90 minutes next week is a good way to start. 

I didn't find the notes slowing me down too much, but it might be good to have the moderator and the note taker be separate people. Letting people focus on one task seems like a good idea. 

In general, we should try to resolve issues in the repo and only pull them into the meeting if they get stuck. So, it should be an exception case to review an issue in a TSC meeting rather than the norm.

**Comment by nzakas** [comment]
I think we did a better job this time, so I think we can close this.
